src/transforms/Rank.js

```diff
@@ -0,0 +1,74 @@
+var Tuple = require('vega-dataflow').Tuple,
+    log = require('vega-logging'),
+    Transform = require('./Transform'),
+    BatchTransform = require('./BatchTransform');
+
+function Rank(graph) {
+  BatchTransform.prototype.init.call(this, graph);
+  Transform.addParameters(this, {
+    field: {type: 'field', default: null},
+    normalize: {type: 'value', default: false}
+  });
+
+  this._output = {
+    'rank': 'rank'
+  };
+
+  return this.mutates(true);
+}
+
+var prototype = (Rank.prototype = Object.create(BatchTransform.prototype));
+prototype.constructor = Rank;
+
+prototype.batchTransform = function(input, data) {
+  log.debug(input, ['rank']);
+
+  var rank  = this._output.rank,
+      norm  = this.param('normalize'),
+      field = this.param('field').accessor,
+      l = {}, i = 0, len = data.length, r, v, d;
+
+  for (; i<len; ++i) {
+    r = norm ? (i+1)/len : i+1;
```

Comment:
Is this supposed to be divided by len or by the number of distinct values? If you have a dataset like [a,b,c,a,b,c], then the ranks are [0.166666, 0.3333333, 0.5, 0.166666, 0.3333333, 0.5]. I mean I can work with this, I just wonder whether this is intended behavior.